core/src/test/java/org/elasticsearch/search/suggest/SuggesterParsingTests.java

```diff
@@ -0,0 +1,136 @@
+import org.elasticsearch.indices.query.IndicesQueriesRegistry;
+import org.elasticsearch.script.ScriptService;
+import org.elasticsearch.search.SearchModule;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.threadpool.ThreadPoolModule;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Tests conversion from X-Content to a SuggestBuilder.
+ */
+public final class SuggesterParsingTests extends ESTestCase {
+
+    private static Injector injector;
+    private static NamedWriteableRegistry namedWriteableRegistry;
+    private static IndicesQueriesRegistry queriesRegistry;
+    private static ParseFieldMatcher parseFieldMatcher;
+    private static Suggesters suggesters;
+
+    /**
+     * Setup for the whole base test class.
+     */
+    @BeforeClass
+    public static void init() throws IOException {
+        Settings settings = Settings.settingsBuilder().put("node.name", AbstractQueryTestCase.class.toString())
+                                .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir())
+                                .put(ScriptService.SCRIPT_AUTO_RELOAD_ENABLED_SETTING.getKey(), false).build();
+        SettingsModule settingsModule = new SettingsModule(settings);
+        namedWriteableRegistry = new NamedWriteableRegistry();
+        injector = new ModulesBuilder().add(new EnvironmentModule(new Environment(settings)), settingsModule,
+            new ThreadPoolModule(new ThreadPool(settings)),
+            new IndicesModule() {
+                @Override
+                protected void configure() {
+                    bindMapperExtension();
+                }
+            }, new SearchModule(settings, namedWriteableRegistry) {
+                @Override
+                protected void configureSearch() {
+                    // Skip me
+                }
+                @Override
+                protected void configureAggs(IndicesQueriesRegistry queriesRegistry) {
+                    // Skip me
+                }
+            }, new AbstractModule() {
+                @Override
+                protected void configure() {
+                    bind(NamedWriteableRegistry.class).toInstance(namedWriteableRegistry);
+                }
+            }).createInjector();
+        queriesRegistry = injector.getInstance(IndicesQueriesRegistry.class);
+        suggesters = injector.getInstance(Suggesters.class);
+        parseFieldMatcher = ParseFieldMatcher.STRICT;
+    }
+
+    @AfterClass
+    public static void afterClass() throws Exception {
+        terminate(injector.getInstance(ThreadPool.class));
+        injector = null;
+        namedWriteableRegistry = null;
+    }
+
+    public void testMalformedJsonRequestPayload() throws Exception {
```

Comment:
I would prefer to add this test to the existing SuggestBuilderTests and not create a new test class for this. If I understand the existing test in CompletionSuggestSearchIT correctly this is only testing pasting, so it should work, or am I missung something?